engine.py

```diff
@@ -45,22 +45,24 @@ class BdfCallbacks:
     def __init__(self, registry):
         self.registry = registry
 
-    def check(self, name, property_name):
+    def check(self, name, property_name="condition_met"):
         obj = self.registry.get_instance(name)
         obj.activate()
+        print("checking " + name + ": " + str(obj.is_true(property_name)))  # debug
```

Comment:
no debugs print should be commited. We need to add a proper logging library to the project, will do that next.